sdk/messaging/azeventhubs/internal/amqpwrap/amqp_error.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package amqpwrap
+
+import (
+	"errors"
+)
+
+// Error is a wrapper that has the context of which connection and
+// link the error happened with.
+type Error struct {
+	ConnID   uint64
+	LinkName string
+	Err      error
+}
+
+func (e Error) Error() string {
+	return e.Err.Error()
+}
+
+func (e Error) As(target any) bool {
+	return errors.As(e.Err, target)
+}
+
+func (e Error) Is(target error) bool {
+	return errors.Is(e.Err, target)
+}
+
+func NewError(err error, connID uint64, linkName string) error {
```

Comment:
I wonder if this would be better named something like `WrapError` since it might not do anything if `err` is `nil` ("New" makes me think it always creates something).  Relatively minor considering this is internal.